libclc/amdgcn-amdhsa/libspirv/misc/sub_group_shuffle.cl

```diff
@@ -8,16 +8,15 @@
 
 #include <spirv/spirv.h>
 
-#define SELF __builtin_amdgcn_mbcnt_hi(-1, __builtin_amdgcn_mbcnt_lo(-1, 0))
```

Comment:
Why do you move away from using `mbcnt`, the trick with feeding it with `-1` seems more efficient than all the calculations done in `__spirv_SubgroupLocalInvocationId`, or perhaps we could use it in there.

From the docs:
> GCN provides a special op-code for compaction within a wavefront – mbcnt. mbcnt(v) computes the bit count of the provided value up to the current lane id. That is, for lane 0, it simply returns popcount (v & 0), for lane 1, it’s popcount (v & 1)

Reply:
I've changed the `__spirv_SubgroupLocalInvocationId` to use this